frontend/src/app/data-entry/gift-form/gift-identification-form/gift-identification-form.component.ts

```diff
@@ -0,0 +1,105 @@
+import { FormGroup, FormControl, Validators } from "@angular/forms";
+import { ActivatedRoute } from "@angular/router";
+import { ToastService } from "@services/toast.service";
+import {
+    DataEntryGiftIdentificationGQL,
+    DataEntryUpdateGiftGQL,
+} from "generated/graphql";
+import {
+    map,
+    switchMap,
+    shareReplay,
+    filter,
+    debounceTime,
+    withLatestFrom,
+} from "rxjs";
+
+@Component({
+    selector: "lc-gift-identification-form",
+    templateUrl: "./gift-identification-form.component.html",
+    styleUrls: ["./gift-identification-form.component.scss"],
+})
+export class GiftIdentificationFormComponent implements OnInit {
+    public id$ = this.route.params.pipe(map((params) => params["id"]));
+
+    public gift$ = this.id$.pipe(
+        switchMap((id) => this.giftQuery.watch({ id }).valueChanges),
+        map((result) => result.data.giftDescription),
+        shareReplay(1)
+    );
+
+    public form = new FormGroup({
+        name: new FormControl("", {
+            validators: [Validators.required],
+            nonNullable: true,
+        }),
+        description: new FormControl("", {
+            nonNullable: true,
+        }),
+    });
+
+    constructor(
+        private destroyRef: DestroyRef,
+        private route: ActivatedRoute,
+        private toastService: ToastService,
+        private giftQuery: DataEntryGiftIdentificationGQL,
+        private giftMutation: DataEntryUpdateGiftGQL
+    ) {}
+
+    ngOnInit(): void {
+        this.gift$
+            .pipe(takeUntilDestroyed(this.destroyRef))
+            .subscribe((gift) => {
+                if (!gift) {
+                    return;
+                }
+                this.form.patchValue(gift, {
+                    emitEvent: false,
+                    onlySelf: true,
+                });
+            });
+
+        this.gift$
+            .pipe(
+                switchMap(() =>
+                    this.form.valueChanges.pipe(
+                        map(() => this.form.getRawValue()),
+                        filter(() => this.form.valid),
+                        debounceTime(300),
+                        withLatestFrom(this.id$),
+                        switchMap(([gift, id]) =>
+                            this.giftMutation.mutate(
+                                {
+                                    giftData: {
+                                        ...gift,
+                                        id,
+                                    },
+                                },
+                                {
+                                    update: (cache) => {
+                                        cache.evict({
+                                            id: `GiftDescriptionType:${id}`,
+                                        });
+                                        cache.gc();
+                                    },
+                                }
+                            )
+                        )
```

Comment:
For the sake of readability, please outfactor the `this.giftMutation.mutate` call to its own method.

Reply:
Done!